packages/angular_devkit/core/src/workspace/json/reader_spec.ts

```diff
@@ -28,7 +28,7 @@ const basicFile = stripIndent`
   "x-bar": 5,
 }`;
 
-const representativeFile = readFileSync(__dirname + '/test/angular.json', 'utf8');
+const representativeFile = readFileSync(require.resolve(__dirname + '/test/angular.json'), 'utf8');
```

Comment:
In general, any sort of node file lookup on Bazel should be subject to `require.resolve`. This is how `rules_nodejs` resolves paths using the Bazel runfiles mechanism, where a given Bazel target only has access to outputs from its dependencies.

In practice, this does not make a lot of difference on Linux. A symlink forest is laid down where the target is going to actually run, and mostly the files are resolved correctly whether you use `require.resolve` or not because the files are there.

On windows though, that's a stricter. Bazel does not lay down a symlink forest on windows by default. If you don't use `require.resolve`, it's still possible to correctly resolve some files, like outputs from other rules. But other files, like node modules dependencies and data files, need to be looked up in the runfiles. 

Since the requirement is quite lax on Linux but quite strict on windows, what ends up happening is that lack of `require.resolve` calls go unnoticed until someone tries to run things on Windows. Then it breaks.

Reply:
Ah. ok. so this is really a workaround for Bazel Windows support?

Since this is not needed on any OS outside of Bazel nor with Bazel on linux/Mac, might be good to add a comment so someone reading through knows why it's there and doesn't try to remove it unintentionally.

Reply:
@filipesilva, this doesn't seem correct for Windows either. Under Bazel to resolve files from the manifest `require.resolve` need to contain a full path including the workspace name, similar to `rlocation`.

Example;
```
require.resolve('workspace_name/tsconfig.json');
````

Will resolve the file from manifest to an absolute path:
```
C:/git/workspace_name/tsconfig.json
```
 
In the above case, the path provided will never be in the MANIFEST, because Bazel paths/keys are never absolute.

MANIFEST extract
```
angular_bazel_architect/src/polyfills.ts C:/git/rules_nodejs/examples/angular_bazel_architect/src/polyfills.ts
angular_bazel_architect/src/styles.scss C:/git/rules_nodejs/examples/angular_bazel_architect/src/styles.scss
angular_bazel_architect/tsconfig.app.json C:/git/rules_nodejs/examples/angular_bazel_architect/tsconfig.app.json
angular_bazel_architect/tsconfig.json C:/git/rules_nodejs/examples/angular_bazel_architect/tsconfig.json
```

Reply:
That's a good point. I wonder why this worked then... I assumed the paths were in the manifest, but didn't verify. 

I know rules_nodejs' `require.resolve` has fallbacks. The absolute path must hit the fallback then? But then this still gets confusing because the data files can only be looked up by runfiles...

Went looking at the require patches in rules_nodejs and found this:
https://github.com/bazelbuild/rules_nodejs/blob/cd8520d870abe490d7a154f7e1d9ccac32ea04d8/internal/node/require_patch.js#L141-L148
```
  // Determine bin and gen root to convert absolute paths into runfile paths.
  const binRootIdx = manifestPath.indexOf(BIN_DIR);
  let binRoot, genRoot;
  if (binRootIdx !== -1) {
    const execRoot = manifestPath.slice(0, binRootIdx);
    binRoot = `${execRoot}${BIN_DIR}/`;
    genRoot = `${execRoot}${GEN_DIR}/`;
  }
```

So that makes sense now: the absolute paths are converted.

Reply:
Thanks for checking @filipesilva.

Where the conversion happens:
https://github.com/bazelbuild/rules_nodejs/blob/cd8520d870abe490d7a154f7e1d9ccac32ea04d8/internal/node/require_patch.js#L307-L315

Reply:
@clydin I'm a bit wary of adding a comment here specifically, because this is just standard practice on rules_nodejs. I agree it would be helpful to have that information somewhere though.

I added it to `docs/process/bazel.md`. We already had some general bazel information there. Does that sound good to you?